### DISCUSSION TITLE
Pass useFieldArray Append function to component

### DISCUSSION THREAD

**Discussion by bjrice88** [react-hook-form/react-hook-form#8779] [Q&A]
In my project I have my form set up so that you can click a button to display a list of objects to add to a form field array.  To accomplish this I have been passing the `append` function that is returned from `useFieldArray` like this.

```
export interface FormPickerProps {
  onCloseHandler: () => void;
  append: UseFieldArrayReturn["append"];
}
```

This worked great up until the most recent update where I am getting a typescript error.
```
Type 'UseFieldArrayAppend<QuoteInput, "forms">' is not assignable to type 'UseFieldArrayAppend<FieldValues, string>'.
```
Is there another way I can define the type of the Append function?  I don't want to define it with the specific generics because I use this function in a variety of places with different forms so I can't lock it down to one specific forms typing.

**Comment by Moshyfawn** [comment]
Even if you're building a generic field array component, you still need to pass the specific to the use-case value types as its generics; you don't need to specify the exact types. Something like this
```ts
import type { UseFieldArrayMethods, FieldValues } from "react-hook-form"

type ComponentProps<
  TFieldValues extends FieldValues,
  TKeyName extends string
> = {
  append: UseFieldArrayMethods<TFieldValues, TKeyName>["append"];
};

const Component = <
  TFieldValues extends FieldValues,
  TKeyName extends string
>({
  append
}: ComponentProps<TFieldValues, TKeyName>) => {
  // INFO: component body
};
```
